var/spack/repos/builtin/packages/py-cupy/package.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyCupy(PythonPackage):
+    """CuPy is an open-source array library accelerated with
+    NVIDIA CUDA. CuPy provides GPU accelerated computing with
+    Python. CuPy uses CUDA-related libraries including cuBLAS,
+    cuDNN, cuRand, cuSolver, cuSPARSE, cuFFT and NCCL to make
+    full use of the GPU architecture."""
+
+    homepage = "https://cupy.dev/"
+    git      = "https://github.com/cupy/cupy"
+
+    version('8.0.0', tag='v8.0.0', submodules=True)
```

Comment:
Git checkouts aren't checksummable. Would it be possible to compile from the release tarballs on PyPI, or are the submodules required?

Reply:
Submodules are required. I don't know if they are included in the pypi tarballs.

Reply:
Can you check?